starter-code/src/viking.js

```diff
@@ -1,11 +1,112 @@
 // Soldier
-function Soldier() {}
+function Soldier(healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg;
+}
+
+
+Soldier.prototype.attack = function(){
+    return this.strength;
+};
+Soldier.prototype.receiveDamage = function(damage){
+    this.health = this.health - damage;
+};
 
 // Viking
-function Viking() {}
+function Viking(name, healthArg, strengthArg) {
+    Soldier.call(this, healthArg, strengthArg);
+    this.name = name;
+}
+
+
+
+Viking.prototype.attack = function(){
+    return this.strength;
+};
+
+
+
+Viking.prototype.receiveDamage = function (damage){
+    this.health = this.health - damage ;
+    if (this.health > 0){
+        return (this.name + " has received " + damage + " points of damage")
+    } else{
+        return (this.name + " has died in act of combat");
+    }
+}
+
+Viking.prototype.battleCry = function(){
+    return "Odin Owns You All!";
+} 
 
 // Saxon
-function Saxon() {}
+function Saxon(healthArg, strengthArg) {
+    Soldier.call(this, healthArg, strengthArg);
+}
+
+Saxon.prototype.attack = function(){
+    return this.strength;
+};
+
```

Comment:
And this one ;) 


```js

Saxon.prototype = Object.create(Soldier.prototype);

```

Reply:
This is how your Saxon and your Viking elements are inheriting from the Soldier constructor